application/ui/src/features/annotator/annotations/annotation-shape-with-labels.component.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+import polylabel from 'polylabel';
+import { useSecondaryToolbarState } from 'src/features/dataset/media-preview/secondary-toolbar/use-secondary-toolbar-state.hook';
+
+import { Annotation, Polygon } from '../types';
+import { AnnotationLabels } from './annotation-labels.component';
+import { getFormattedPoints } from './utils';
+
+type AnnotationShapeProps = {
+    annotation: Annotation;
+};
+export const AnnotationShapeWithLabels = ({ annotation }: AnnotationShapeProps) => {
+    const { shape, labels } = annotation;
+    const color = labels.length ? labels[0].color : '--annotation-fill';
+    const { removeLabels } = useSecondaryToolbarState();
+
+    if (shape.type === 'rectangle') {
+        return (
+            <g transform={`translate(${shape.x}, ${shape.y})`}>
+                <rect aria-label='annotation rect' x={0} y={0} width={shape.width} height={shape.height} fill={color} />
```

Comment:
Can we use the `AnnotationShape` for the acutal shape? (i.e. for both the rect and polygon)
That way we avoid styling differences.